python/lsst/ctrl/mpexec/cmdLineFwk.py

```diff
@@ -353,11 +353,21 @@ def makeWriteButler(cls, args: argparse.Namespace) -> Butler:
             chainDefinition = list(self.output.chain if self.output.exists else self.inputs)
             if args.replace_run:
                 replaced, _ = chainDefinition.pop(0)
-                if args.prune_replaced:
-                    # TODO: DM-23671: need a butler API for pruning an
-                    # entire RUN collection, then apply it to 'replaced'
-                    # here.
-                    raise NotImplementedError("Support for --prune-replaced is not yet implemented.")
+                if args.prune_replaced == "unstore":
+                    # Remove datasets from datastore
+                    with butler.transaction():
+                        refs = butler.registry.queryDatasets(..., collections=replaced, deduplicate=True)
```

Comment:
Adding `deduplicate` when there's only one collection is unnecessary - that just invokes the "return the dataset from the first collection in the ordered list" logic, rather than doing a SELECT DISTINCT or something.  It should be harmless, but I should make sure it's truly a no-op (rather than a more complex query for no reason) when there is only one collection; I'm not sure that's the case now.  But I see that `Butler.pruneCollection` does this too, so it's probably my fault for giving you a bad example to learn from.

I wonder if it would be better to modify `Butler.pruneCollection` to accept `unstore=True, purge=False` on a `RUN` so we could use that here; I think that could reduce the amount of total logic this operation goes through.

Reply:
I removed `deduplicate`, indeed I was pasting without much thinking here.

Reply:
For `pruneCollection` update I'm also happy to defer to a next ticket.